### PR TITLE
grafana datasource proxy

### PR DESCRIPTION
**Pull request by langerma**
i added the ability to use grafana datasource proxy

### CODE REVIEW

File: check_graphite [hunk 1]
```diff
@@ -31,20 +33,28 @@ EXIT_WARNING    = 1
...
 
+requests.packages.urllib3.disable_warnings()
+
 def check(host, metric, warning_threshold, critical_threshold,
-          invert, function, duration, user, password):
+          invert, function, duration, user, password, grafana, datasourceid, authkey):
 
-    if not host.startswith('http'):
-        host = 'http://%s' % host
+    if not host.startswith('https'):
```
Comment (andreasjansson) [inline check_graphite]: why this change?

File: check_graphite [hunk 2]
```diff
@@ -31,20 +33,28 @@ EXIT_WARNING    = 1
...
     if warning_threshold:
         warning_threshold = convert_si(warning_threshold)
 
     if warning_threshold is not None and warning_threshold > critical_threshold:
         invert = True
-
-    url = '%s/render' % host
+    if grafana == "yes":
+        url = str(host) + '/api/datasources/proxy/' + str(datasourceid) + '/render'
+        print url
```
Comment (andreasjansson) [inline check_graphite]: remove print statements

File: check_graphite [hunk 3]
```diff
@@ -54,15 +64,24 @@ def check(host, metric, warning_threshold, critical_threshold,
...
-        r = requests.get(url, params=params, auth=auth)
-    except requests.exceptions.ConnectionError, e:
-        critical('Failed to contact Graphite server', e.message)
-    try:
-        data = r.json()
-    except json.JSONDecodeError, e:
-        critical('Could not decode JSON', r.content)
+    if grafana=="yes":
+        try:
+            r = requests.get(url, headers=auth_header, params=params, verify=False)
```
Comment (andreasjansson) [inline check_graphite:64]: why not just wrap this line with the grafana check? the rest is the same in both branches.

File: check_graphite [hunk 4]
```diff
@@ -157,23 +176,28 @@ def convert_si(value):
...
-    parser.add_argument('--metric', '-m', required=True, help='Metric to check, e.g. web01.processes.fork_rate')
-    parser.add_argument('--critical', '-c', required=True, help='Critical threshold, e.g. 1.0, 3G')
-    parser.add_argument('--warning', '-w', help='Warning threshold, e.g. -1.1k, 20')
-    parser.add_argument('--invert', '-i', action='store_true', help='Invert the check so that an alert is triggered if the value falls below the threshold. Invert is implied if warning threshold > critical threshold')
-    parser.add_argument('--duration', '-d', type=int, default=5, help='Number of minutes of data to aggregate')
-    parser.add_argument('--function', '-f', default='average', choices=['average', 'max', 'min', 'sum'], help='The aggregation function to apply to the time series')
-    parser.add_argument('--user', '-u', help='HTTP auth user')
-    parser.add_argument('--password', '-p', help='HTTP auth password')
-
+    parser.add_argument('--host',         '-o', required=True, help='Graphite host, e.g. graphite.example.com, https://1.2.3.4, graphite01:8080')
```
Comment (andreasjansson) [inline check_graphite]: Prefer not to add extraneous space for alignment as per PEP8.

File: check_graphite [hunk 5]
```diff
@@ -157,23 +176,28 @@ def convert_si(value):
...
-    check(host=args.host,
-          metric=args.metric,
-          warning_threshold=args.warning,
-          critical_threshold=args.critical,
-          invert=args.invert,
-          function=args.function,
-          duration=args.duration,
-          user=args.user,
-          password=args.password)
+    check(host               = args.host,
```
Comment (andreasjansson) [inline check_graphite]: Prefer not to add spaces for alignment.

File: check_graphite [hunk 6]
```diff
@@ -157,23 +176,28 @@ def convert_si(value):
...
+    parser.add_argument('--metric',       '-m', required=True, help='Metric to check, e.g. web01.processes.fork_rate')
+    parser.add_argument('--critical',     '-c', required=True, help='Critical threshold, e.g. 1.0, 3G')
+    parser.add_argument('--warning',      '-w', help='Warning threshold, e.g. -1.1k, 20')
+    parser.add_argument('--invert',       '-i', action='store_true', help='Invert the check so that an alert is triggered if the value falls below the threshold. Invert is implied if warning threshold > critical threshold')
+    parser.add_argument('--duration',     '-d', type=int, default=5, help='Number of minutes of data to aggregate')
+    parser.add_argument('--function',     '-f', default='average', choices=['average', 'max', 'min', 'sum'], help='The aggregation function to apply to the time series')
+    parser.add_argument('--user',         '-u', help='HTTP auth user')
+    parser.add_argument('--password',     '-p', help='HTTP auth password')
+    parser.add_argument('--grafana',      '-g', default='no', choices=['yes', 'no'], help='use grafana datasource proxy')
```
Comment (andreasjansson) [inline check_graphite]: `--grafana` could be a boolean flag. Also uppercase first letter of help.

File: check_graphite (hunk 3 above)
Comment (andreasjansson) [inline check_graphite:64]: also, move this line to where the auth_header, etc., is defined. So for example

```
params = ...
try:
    if grafana:
        url = str(host) + '/api/datasources/proxy/' + str(datasourceid) + '/render'
        auth_header = {'Authorization': 'Bearer ' + authkey}
        r = requests.get(url, headers=auth_header, params=params, verify=False)
    else:
        ...
```

File: check_graphite [hunk 7]
```diff
@@ -4,6 +4,7 @@
 
 check_graphite
 Copyright (C) 2014  Andreas Jansson
+Copyright (C) 2016  Markus  Langer
```
Comment (andreasjansson) [inline check_graphite]: Let's change this to

```
Copyright (C) 2016  The project contributors
```

**Comment by andreasjansson** [comment]
Please squash commits

File: check_graphite (hunk 1 above)
Comment (langerma) [inline check_graphite]: because i wanted to use https.

Comment (andreasjansson) [inline check_graphite]: since this is a breaking change, I'd prefer if we leave the default as http and let users type out "https://" in the host if they want to use https. the `host.startswith('http')` check will allow `https`, but changing the check to `startswith('https')` would prepend `https://http://` if the host already starts with `http://`.

Comment (langerma) [inline check_graphite]: right that was my fault. sorry for that. i'll correct that later at home.

**Comment by langerma** [comment]
a sorry i just read about squashing commits, next time i am contributing i am doing it in a feature branch and then squashing all commits, before opening a pull req

**Comment by andreasjansson** [comment]
You can still squash them
http://stackoverflow.com/questions/5667884/how-to-squash-commits-in-git-after-they-have-been-pushed

but not a big deal really

**Comment by langerma** [comment]
sorry rebasing does not work
